block/manager.go

```diff
@@ -245,31 +283,59 @@ func (m *Manager) AggregationLoop(ctx context.Context, lazy bool) {
 	}
 }
 
+// BlockSubmissionLoop is responsible for submitting blocks to the DA layer.
+func (m *Manager) BlockSubmissionLoop(ctx context.Context) {
+	timer := time.NewTicker(m.conf.DABlockTime)
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-timer.C:
+			err := m.submitBlocksToDA(ctx)
+			if err != nil {
+				m.logger.Error("error while submitting block to DA", "error", err)
+			}
+		}
+	}
+}
+
 // SyncLoop is responsible for syncing blocks.
 //
-// SyncLoop processes headers gossiped in P2p network to know what's the latest block height,
+// SyncLoop processes headers gossiped in P2P network to know what's the latest block height,
 // block data is retrieved from DA layer.
 func (m *Manager) SyncLoop(ctx context.Context, cancel context.CancelFunc) {
 	daTicker := time.NewTicker(m.conf.DABlockTime)
+	blockTicker := time.NewTicker(m.conf.BlockTime)
 	for {
 		select {
 		case <-daTicker.C:
 			m.retrieveCond.Signal()
+		case <-blockTicker.C:
+			m.blockStoreCond.Signal()
 		case blockEvent := <-m.blockInCh:
 			block := blockEvent.block
 			daHeight := blockEvent.daHeight
+			blockHash := block.Hash().String()
 			m.logger.Debug("block body retrieved from DALC",
 				"height", block.SignedHeader.Header.Height(),
 				"daHeight", daHeight,
-				"hash", block.Hash(),
+				"hash", blockHash,
 			)
-			m.syncCache[block.SignedHeader.Header.BaseHeader.Height] = block
+			if m.blockCache.isSeen(blockHash) {
+				m.logger.Debug("block already seen", "height", block.Height(), "block hash", blockHash)
+				continue
+			}
+			m.blockCache.setBlock(block.SignedHeader.Header.BaseHeader.Height, block)
```

Comment:
we are referencing the height in 3 different ways here. 

```
block.SignedHeader.Header.Height()
block.Height()
block.SignedHeader.Header.BaseHeader.Height
```

If these are all the same, that is super confusing and we should figure out a way to potentially clean that up, but that is a separate issue.  But we should pick on, I'd recommend `block.Height()` and assign a `blockHeight` variable in the beginning. 

If these are not the same, then we should add more clarify, either with comments or with better naming, as to what the difference is.

Reply:
Yes, these are all the same. I assigned a `blockHeight` variable in the beginning and made the same change in other places where this existed. Thanks!